Core/src/IfcGeometryConverter/PlacementConverter.h

```diff
@@ -754,97 +754,152 @@ namespace OpenInfraPlatform {
+							if (GeomUtils::LineSegmentToLineSegmentIntersection(xAxisPosition2, xAxisPosition, yAxisPosition2, yAxisPosition, intersectionPoint)) {
+								i = xAxis.size();
+								j = yAxis.size();
 							}
 						}
 					}
 					
-					//orthogonal complement of the IntersectingAxes[1]
-					//	<x,y> = 0
-					//	x1y1 + x2y2 = 0
-					//	y2 = -x1y1/x2
-					carve::geom::vector<2> IntersectingAxes = X_Axis_2Position - X_Axis_Position;
-					carve::geom::vector<2> Orthogonal_complement;
-					if (IntersectingAxes[0] != 0.0 && IntersectingAxes[1] != 0.0) {
-						Orthogonal_complement[0] = 1.0;
-						Orthogonal_complement[1] = -IntersectingAxes[0] * Orthogonal_complement[0] / IntersectingAxes[1];
-					}
-					else if (IntersectingAxes[0] = 0.0) {
-						Orthogonal_complement[0] = 1.0;
-						Orthogonal_complement[1] = 0.0;
-					}
-					else {
-						Orthogonal_complement[0] = 0.0;
-						Orthogonal_complement[1] = 1.0;
-					}
-
-					//cross product of IntersectingAxes[1] and the orthogonal complement of the IntersectingAxes[1] 
-					double cross_product = (IntersectingAxes[0] * Orthogonal_complement[1]) - (IntersectingAxes[1] * Orthogonal_complement[0])
-
-					carve::geom::vector<3> location = {0.0,0.0,0.0};
-					location[0] = intersection_point[0];
-					location[1] = intersection_point[1];
-					location[2] = cross_product;
-
+					carve::geom::vector<2> intersectingAxes1 = xAxisPosition2 - xAxisPosition;
+					carve::geom::vector<2> intersectingAxes2 = yAxisPosition2 - yAxisPosition;
+
+					carve::geom::vector<3> location = carve::geom::VECTOR(0.0,0.0,0.0);
+					carve::geom::vector<2> newIntersectionPoint = calculateIntersectionPointWithOffsets(intersection, xAxisPosition, xAxisPosition2, yAxisPosition, yAxisPosition2);
+					carve::geom::vector<3> crossProduct;
+					carve::geom::vector<3> intersecting3DAxes1;
+					carve::geom::vector<3> orthogonalComplement;
 					switch (intersection->OffsetDistances.size())
 					{
 					case 3:
-						double Xdistance = intersection->OffsetDistances[0];
-						double Ydistance = intersection->OffsetDistances[1];
-						double Zdistance = intersection->OffsetDistances[2];
-						location[0] = location[0] + intersection->OffsetDistances[1];
-						location[1] = location[1] + intersection->OffsetDistances[0];
-						location[2] = location[2] + intersection->OffsetDistances[2];
+						intersecting3DAxes1 = carve::geom::VECTOR(intersectingAxes1.x, intersectingAxes1.y, 0.0 );
+						orthogonalComplement = carve::geom::VECTOR(-intersectingAxes1.y, intersectingAxes1.x, 0.0);
+						
+						//cross product of IntersectingAxes[1] and the orthogonal complement of the IntersectingAxes[1] 
+						crossProduct = carve::geom::cross(intersecting3DAxes1, orthogonalComplement);
+						location = location + crossProduct;
+						
+						location.x = location.x + newIntersectionPoint.x;
+						location.y = location.y + newIntersectionPoint.y;
+						location.z = location.z + intersection->OffsetDistances[2];

```

Comment:
This is wrong. I see that we need to make a small excursion to vector algebra.

Reply:
Resolved: 4dc4fbf